opentelemetry-otlp/src/exporter/mod.rs

```diff
@@ -92,6 +93,44 @@ impl Default for ExportConfig {
     }
 }
 
+#[derive(Error, Debug)]
+/// Errors that can occur while building an exporter.
+// TODO: Refine and polish this.
+// Non-exhaustive to allow for future expansion without breaking changes.
+// This could be refined after polishing and finalizing the errors.
+#[non_exhaustive]
+pub enum ExporterBuildError {
+    /// Spawning a new thread failed.
+    #[error("Spawning a new thread failed. Unable to create Reqwest-Blocking client.")]
+    ThreadSpawnFailed,
```

Comment:
This will only happens if reqwest-blocking is used right? If so can we guard it in feature flag?

Reply:
yes, thats why I put a big TODO on this. we definitely need to refine it and see which ones makes sense and remove others.